Reload of mTLS certificates on SIGHUP. More graceful shutdown on SIGINT or SIGTERM

piotrsuchy opened the pull request:
## Description

This PR makes gobgp_exporter support reloading of mTLS certificates on SIGHUP signal.
It also adds a more graceful shutdown on receiving SIGINT or SIGTERM signal - it waits until connections get closed OR 10 seconds.

### Other things

- creating tlsutils package and moving tls related things there from main.go
- adding / unifying some logs
- removing some unused functions or obsolete nil != nil check in collect_peers.go
- adding .vscode to .gitignore

### How cert reload is implemented here

TLSReloader struct holds all cert related information based on which http server has mtls: certPath, keyPath, caPath... Using tlsReloader.Reload() updates pointer to Certificate and pointer to caPool. It is only done for https server, http doesn't need it ofc. Separate goroutine for http/https server and signal handling.

### How it was tested

1. Testing reloading: Get a bash script to generate client and server certs, keys and cacerts. After generating the first time they match. Curl works:

```
curl --cacert ca.crt --cert client.crt --key client.key https://localhost:9474/metrics
```

Generate certs again and now client has new ones, server doesn't have them updated. Curl doesn't work:

```
2024/12/02 05:25:59 http: TLS handshake error from [::1]:44760: local error: tls: bad record MAC
```

Send `kill -SIGHUP $(pidof gobgp_exporter)` and try curling again. We get:

```
INFO[2024-12-02 05:26:06] source: main.go:229 // Received SIGHUP, reloading certificates
INFO[2024-12-02 05:26:06] source: main.go:233 // Certificates reloaded
```

And curl works again. If there is an error during reload it throws an error but won't shut down the whole process.

2. Testing graceful shutdown - sending `kill -SIGINT/SIGTERM $(pidof gobgp_exporter)` -> it shuts down after 10s, if there is a connection open (tested by adding a 30 s sleep to HandleFunc("/") for example). it shuts down instantly if there is no connection open. It shuts down after a request is done if it's closed faster than the duration of 10s.

kacpekwasny commented on pkg/tlsutil/mtls_reload.go:
```diff
@@ -1 +1,70 @@
+		keyPath:  keyPath,
+		caPath:   caPath,
+	}
+	// reload works for a first time load as well
```
just my thought i wanted to share, I reallllly dont need this to be changed
 one might want to create the TLSreloader before the files are on disk, then you cannot use this utility function, so my prefered style would not to have Reload withing NewTLSReloader.

piotrsuchy replied:
sure, updated

kacpekwasny reviewed (approved):
wow, GetCertificate is cool way to reload certs, nice find

piotrsuchy commented:
Fixed a missing check in serverShutdownWithTimeout and added a .golangci.yaml file that help with a linter. Removed unused function (that wasn't ever used I think it was debug code that got accidently merged). 
Did some more testing and it works as expected, when changing CA it needs to be reloaded, when CA is the same, just new certs they keep on working.
